mypy/main.py

```diff
@@ -482,6 +528,33 @@ def add_invertible_flag(flag: str,
         print("Warning: --no-fast-parser no longer has any effect.  The fast parser "
               "is now mypy's default and only parser.")
 
+    try:
```

Comment:
This is a pretty complex block of logic -- can you at least split it out into a helper function?